packages/donor/src/screens/bookDonation/BookDonationScreenContainer.tsx

```diff
@@ -29,26 +31,47 @@ export function BookDonationScreenContainer({
     return <Navigate to={MainNavigationKeys.Approve} />;
   }
 
-  if (isLoggedIn && appointmentToBookStore.hasAppointmentToBook()) {
+  if (
+    isLoggedIn &&
+    appointmentToBookStore.hasAppointmentToBook() &&
+    appointmentToBookStore.isDonationValidForDonor(appState.donor!)
+  ) {
     return <Navigate to={MainNavigationKeys.Questionnaire} />;
   }
 
   const onSlotSelected = (donationSlot: DonationSlotToBook) => {
     appointmentToBookStore.setAppointmentToBook(donationSlot);
+
     if (isLoggedIn) {
-      navigate(MainNavigationKeys.Questionnaire);
+      appointmentToBookStore.isDonationValidForDonor(appState.donor!)
+        ? navigate(MainNavigationKeys.Questionnaire)
+        : setShowWarningPopup(true);
     } else {
       navigate(MainNavigationKeys.Register);
     }
   };
 
+  const onSlotSelectedPopUpProps = {
+    open: showWarningPopup,
+    onApproved: () => navigate(MainNavigationKeys.Questionnaire),
+    onBack: () => {
+      setShowWarningPopup(false);
+      navigate(MainNavigationKeys.BookDonation);
+    },
+    onBClose: () => {
```

Comment:
onClose

Reply:
Also, how is this not a build failure?

Reply:
There is no type and "onClose" was not in the pick @yaron1m

Reply:
I don't understand, how is `onBClose` not a typo?

Reply:
`onSlotSelectedPopUpProps` has no type. 
When it is passed to `BookDonationScreen`, `BookDonationScreen` looks for the props that are in the pick, `onClose` is not in the pick and nobody looks at it

Reply:
So should it be in the pick?

Reply:
Yes, added it